driver/src/main/java/org/neo4j/driver/internal/cluster/RoutingProcedureRunner.java

```diff
@@ -71,16 +76,43 @@ public RoutingProcedureRunner( RoutingContext context )
                 .asyncResult().thenCompose( StatementResultCursor::listAsync );
     }
 
-    private Statement procedureStatement( ServerVersion serverVersion )
+    private DirectConnection connection( ServerVersion serverVersion, Connection connection )
     {
-        if ( serverVersion.greaterThanOrEqual( v3_2_0 ) )
+        if ( serverVersion.greaterThanOrEqual( v4_0_0 ))
```

Comment:
Maybe we should consider moving to compare protocol versions instead of server versions over time.

Reply:
Strictly speaking we cannot test bolt version as the procedures are not part of bolt protocol.